utils/src/main/java/de/hellfish/ngrok/utils/HttpRequestConverter.java

```diff
@@ -0,0 +1,30 @@
+package de.hellfish.ngrok.utils;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.apache.commons.io.IOUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Enumeration;
+
+public class HttpRequestConverter {
+
+    public static HttpRequest convert(HttpServletRequest request) {
+        HttpRequest result = new HttpRequest();
+        Enumeration<String> headerNames = request.getHeaderNames();
+        while (headerNames.hasMoreElements()) {
+            String headerName = headerNames.nextElement();
+            String headerValue = request.getHeader(headerName);
+            result.addHeader(headerName, headerValue);
```

Comment:
It's better to remove `addHeader` from a `HttpRequest` class. Just collect headers separately in this method and pass a Map<String, String> to a constructor of the `HttpRequest` class.